servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/HttpsProxyTest.java

```diff
@@ -101,7 +101,8 @@ void startServer() throws Exception {
     void createClient() {
         assert serverAddress != null && proxyAddress != null;
         client = HttpClients
-                .forSingleAddressViaProxy(serverAddress, proxyAddress)
+                .forSingleAddress(serverAddress)
```

Comment:
For `HttpsProxyTest` and `HttpProxyTest`, can we parametrize at least one test to verify that `forResolvedAddress` with proxy also works?

Reply:
Interesting that you ask. I started parametrizing the tests and was surprised to discover the resolved version did not work:

```
Unexpected address resolution request: 127.0.0.1:55937
java.lang.IllegalArgumentException: Unexpected address resolution request: 127.0.0.1:55937
	at io.servicetalk.http.netty.DefaultSingleAddressHttpClientBuilder$NoopServiceDiscoverer.discover(DefaultSingleAddressHttpClientBuilder.java:667)
```

As a result, I tried to parameterize the test on the `main` branch and it also fails when using `forResolvedAddressViaProxy` in place of `forSingleAddressViaProxy`. The reason is that the `NoopServiceDiscoverer` expects the discovered address to be the same as the address configured originally, while the discovered address is in fact the address of the proxy and not the eventual target.

Do we really need the verification that causes the failure? https://github.com/apple/servicetalk/blob/main/servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java#L665-L667 <- Perhaps we can get rid of this validation?

Reply:
Please see my latest commit. I was really surprised the proxy doesn't work with resolved addresses even without my changes. So I made an attempt to fix it. I got rid of an optimization for caching the `Publisher` in the `NoopServiceDiscoverer` in the constructor and return a new instance upon each `discover`, but I don't think it would cause any performance hit as it's supposed to be executed once upon client initialization. Internal APIs needed to be change to facilitate the feature, but now it's possible to use a proxy in any case (except for an unresolved proxy address when the client was created for a resolved address -> for that I added a javadoc note).

Reply:
Agreed, it should not affect perf. Thanks for fixing it!